compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -687,5 +703,13 @@ object LazyVals {
     val current: TermName     = "current".toTermName
     val lock: TermName        = "lock".toTermName
     val discard: TermName     = "discard".toTermName
+    val compareAndSet: TermName = "compareAndSet".toTermName
+    val lzyHandleSuffix: String = "$$lzyHandle"
+    val methodHandleLookupSuffix: String = "$$methodHandleLookup"
   }
+
+  extension (sym: Symbol) def isVarHandleForLazy(using Context) =
+    sym.name.endsWith(lazyNme.lzyHandleSuffix)
```

Comment:
Looks like we should use a dedicated `NameKind` if we need to do that kind of test.